### PR TITLE
v3/iterators_tests

### PR DESCRIPTION
**Pull request by sahils-tenable**
- Updated the automation tests for version 3 iterators covered in #69 
- This PR depends upon #69

### CODE REVIEW

**Review by milanaghera-tenable — APPROVED** [review]

**Review by harshshah-tenable — APPROVED** [review]

File: tests/io/v3/base/test_csv_iterator.py [hunk 1]
```diff
@@ -74,3 +123,19 @@ def test_csv_iterator(api):
     )
     assert next(csv_iterator) == CSV_TEXT
     assert next(csv_iterator) == CSV_TEXT_2
```
Comment (harshshah-tenable) [inline tests/io/v3/base/test_csv_iterator.py:121]: here we would need to register CSV_TEXT_2 in the body as well correct? while registering the response
CC: @milanaghera-tenable

Comment (sahils-tenable) [reply tests/io/v3/base/test_csv_iterator.py:121]: No, this csv_text_2 is being handled by the iterator itself.

File: tests/io/v3/base/test_explore.py [hunk 2]
```diff
@@ -127,6 +127,9 @@
 
 
 def test_search(api):
```
Comment (Mangusingh-te) [inline tests/io/v3/base/test_explore.py:129]: Test method name is bit vague. Rename either test-method name or update docstring.

**Review by Mangusingh-te — COMMENTED** [review]
Add a test-case, which tests, **next** token being sent in next subsequent request, when all records in current page are consumed. [Existing test-case](https://github.com/dhvanilshah-tenable/pyTenable/pull/83/files#diff-5830bbbee39b922670e2a656f9227a21cf1d254ec45edeb9275a25be05c7594aR116) doesn't cover multi-paged response scenario.

**Comment by sahils-tenable** [comment]
> Add a test-case, which tests, **next** token being sent in next subsequent request, when all records in current page are consumed. [Existing test-case](https://github.com/dhvanilshah-tenable/pyTenable/pull/83/files#diff-5830bbbee39b922670e2a656f9227a21cf1d254ec45edeb9275a25be05c7594aR116) doesn't cover multi-paged response scenario.

Hi @Mangusingh-te this logic take care of multi-paged response scenario with next token, there are some minor change required which i'll update. https://github.com/dhvanilshah-tenable/pyTenable/pull/83/files#:~:text=__contains__(next(search_iterator))-,responses.reset(),next(search_iterator_1),-%C2%A9%202022%20GitHub%2C%20Inc

File: tests/io/v3/base/test_explore.py (hunk 2 above)
Comment (sahils-tenable) [reply tests/io/v3/base/test_explore.py:129]: @Mangusingh-te this method is testing the _search method of base class that is why we have name to method ```test_search```. do you have any better suggestion?

**Pull request merged**